plutus-use-cases/src/Language/Plutus/Coordination/Contracts/CrowdFunding.hs

```diff
@@ -18,35 +19,51 @@ module Language.Plutus.Coordination.Contracts.CrowdFunding (
     , collectFundsTrigger
     ) where
 
-import           Language.Plutus.Coordination.Plutus
+import           Control.Applicative                  (Alternative)
+import           Control.Monad                        (Monad (..), guard)
+import qualified Data.Set                             as Set
+import           Language.Plutus.Coordination.Plutus  (Height, PendingTx (..), PendingTxIn (..), PubKey (..), Value)
 import qualified Language.Plutus.CoreToPLC.Primitives as Prim
 import           Language.Plutus.TH                   (plutus)
+import           Wallet.API                           (EventTrigger (..), Range (..), WalletAPI (..), pubKey)
+import           Wallet.UTXO                          (Address', DataScript (..), Tx (..), TxIn (..), TxOut (..),
+                                                       TxOutRef', Validator (..), hashValidator)
+import qualified Wallet.UTXO                          as UTXO
 
-import           Prelude                              (Bool (..), Either (..), Num (..), Ord (..), succ, sum, ($))
+import           Prelude                              (Bool (..), Num (..), Ord (..), fromIntegral, succ, sum, ($),
+                                                       (<$>))
 
 -- | A crowdfunding campaign.
 data Campaign = Campaign
-    { campaignDeadline           :: !BlockHeight
+    { campaignDeadline           :: !Height
     , campaignTarget             :: !Value
-    , campaignCollectionDeadline :: !BlockHeight
-    , campaignOwner              :: !PubKey
+    , campaignCollectionDeadline :: !Height
+    , campaignOwner              :: !CampaignActor
     }
 
+newtype CampaignActor = CampaignActor { campaignActor :: PubKey }
+
 -- | Contribute funds to the campaign (contributor)
 --
-contribute :: Campaign -> Value -> TxM [TxOutRef]
+contribute :: (Alternative m, WalletAPI m, Monad m) => Campaign -> Value -> m ()
```

Comment:
We should make `Alternative` a superclass of `WalletAPI` because there are many operations that can fail, so we could use `<|>` in a meaningful way (to specify alternatives)